graphula-persistent/src/Graphula/Persist.hs

```diff
@@ -1,36 +1,51 @@
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE ConstraintKinds #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE UndecidableInstances #-}
-module Graphula.Persist (persistGraph, onlyKey, keys, Keys, PersistRecord) where
+module Graphula.Persist (GraphulaPersistT, runGraphulaPersistT, onlyKey, keys, Keys, PersistRecord) where
 
 import Graphula
 import Control.Monad.IO.Class
-import Control.Monad.Trans.Reader
+import Control.Monad.Reader
+import Control.Monad.Catch (MonadCatch, MonadThrow, MonadMask)
 import Database.Persist
 import Database.Persist.Sql
 import GHC.TypeLits (TypeError, ErrorMessage(..))
 
-persistGraph
-  :: (SqlBackendCanWrite backend, MonadIO m, MonadIO n)
-  => (forall b. ReaderT backend n b -> m b)
-  -> Frontend (PersistRecord backend) Entity (m a) -> m a
-persistGraph runDB = \case
-  Insert n next -> do
-    x <- runDB $ do
+newtype RunDB backend n m = RunDB (forall b. ReaderT backend n b -> m b)
```

Comment:
Curious why we have both `n` and `m` here and below?

Reply:
Oop, never mind, `withGraph` above is a good example. `runDB` might peel off a bunch of nonsense to get to `m b`